lib/salus/scanners/osv/gradle_osv.rb

```diff
@@ -0,0 +1,170 @@
+
+      report_failure
+      log(JSON.pretty_generate(results))
+    end
+
+    private
+
+    # Match if dependency version found is in the range of
+    # vulnerable dependency found.
+    def version_matching(version, introduced, fixed)
+      vulnerable_flag = false
+      if introduced.present? && fixed.present?
+        if SemVersion.new(version) >= SemVersion.new(introduced) &&
+            SemVersion.new(version) < SemVersion.new(fixed)
+          vulnerable_flag = true
+        end
+      elsif introduced.present?
+        vulnerable_flag = true if SemVersion.new(version) >= SemVersion.new(introduced)
+      end
+
+      vulnerable_flag
+    end
+
+    # Compare vulnerabilities found with dependencies found
+    # and return vulnerable dependencies
+    def match_vulnerable_dependencies(dependencies)
+      results = []
+      dependencies.each do |dependency|
+        lib = "#{dependency['group_id']}:#{dependency['artifact_id']}"
+
+        if dependency['version'].present?
+          version = dependency['version']
+          # Cleanup version string.
+          version = version.delete("^0-9.").gsub(/\.+$/, "")
+          package_matches = @osv_vulnerabilities.select do |v|
+            v.dig("package", "name") == lib
+          end
+
+          package_matches.each do |match|
+            # 'match' format
+            # {
+            #   "package"=>{"name"=>"sample:sample-java", "ecosystem"=>"Maven",
+            #   "purl"=>"pkg:maven/sample/sample-java"},
+            #   "ranges"=>[{"type"=>"ECOSYSTEM", "events"=>[{"introduced"=>"0"},
+            #   {"fixed"=>"8.0.16"}]}], "versions"=>["2.0.14", "3.0.10"],
+            #   "database_specific"=>{"cwe_ids"=>["CWE-000"], "github_reviewed"=>true,
+            #   "severity"=>"MODERATE"}, "id"=>"GHSA-xxxx",
+            #   "summary"=>"Privilege escalation", "details"=>"Vulnerability",
+            #   "aliases"=>["CVE-0000"], "modified"=>"2022-00-00T00:00:00.00Z",
+            #   "published"=>"2020-00-00T00:00:00Z", "references"=>[{"type"=>"ADVISORY",
+            #   "url"=>"https://nvd.nist.gov/vuln/detail/CVE-0000-0000"}],
+            #   "schema_version"=>"1.2.0", "severity"=>[{"type"=>"CVSS_V3",
+            #   "score"=>"CVSS:3.0"}], "database"=>"Github Advisory Database"
+            # }
+            match["ranges"].each do |version_ranges|
+              introduced, fixed = vulnerability_info_for(version_ranges)
+              if %w[SEMVER ECOSYSTEM].include?(version_ranges["type"]) &&
+                  version_matching(version, introduced, fixed)
+                puts "Found", match["package"]["name"]
+                results.append(format_vulnerability_result(match, version, introduced, fixed))
+              end
+            end
+          end
+        end
+      end
+      results
+    end
+
+    # Find dependencies from the project
+    def find_dependencies
+      shell_return = run_shell("/home/bin/parse_gradle_deps")
```

Comment:
Have you had any issues with run_shell("bin/parse_gradle_deps") ?  
Asking as the absolute path will tightly couple this to the dockerfile

Reply:
Was following the similar command used in Gradle Reporting Scanner. Checking to get some more details.